examples/systemcalls/systemcalls.c

```diff
@@ -10,12 +15,55 @@
 bool do_system(const char *cmd)
 {
 
+    int status;
+
 /*
  * TODO  add your code here
  *  Call the system() function with the command set in the cmd
  *   and return a boolean true if the system() call completed with success 
  *   or false() if it returned a failure
 */
+    if (cmd != NULL)
+    {
+
+        status = system(cmd);
+
+        if (status < 0)
+        {
+
+            perror("system");
+            return false;
+
+        }
+        else
+        {
+
+            if (WIFEXITED(status))
+            {
+               if (WEXITSTATUS(status) == 0)
+                {
+                    // do nothing here
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                
+                return false;
+            }
+
+        }
+
+    }
+    else
+    {
+
+        return false;
```

Comment:
@SundarKrishnakumar consider simplifying this function and associated logic by returning true in the "do nothing" path and returning false in the fallthrough and everywhere else, replacing the return true below which is dead code.